src/SonarScanner.MSBuild.Shim/Interfaces/ISonarProjectPropertiesValidator.cs

```diff
@@ -18,15 +18,12 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
-using SonarScanner.MSBuild.TFS;
+using System.Collections.Generic;
 
-namespace SonarScanner.MSBuild
+namespace SonarScanner.MSBuild.Shim.Interfaces
 {
-    public class NullCoverageReportConverter : ICoverageReportConverter
+    public interface ISonarProjectPropertiesValidator
     {
-        public bool ConvertToXml(string inputFilePath, string outputFilePath)
-            => false;
-
-        public bool Initialize() => true;
+        bool AreExistingSonarPropertiesFilesPresent(string sonarScannerCwd, ICollection<ProjectData> projects, out IList<string> invalidFolders);
```

Comment:
Does the output parameter need to be a `IList` rather than an `IEnumerable`? (returning a list implies list-like behaviour is needed by the caller)

Reply:
What are the specific behaviors for IList ? Indeed we're just using this list for a string.join() operation, i think IEnumerable will be sufficient.